### ISSUE TITLE
needless-arbitrary-self-type ignores raw identifiers

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#13899]
### Summary

clippy misses the "#r" in the suggestion which then causes an error 

### Lint Name

needless-arbitrary-self-type

### Reproducer

I tried this code:

```rust
//@ check-pass
//@ edition: 2021

fn foo<'r#struct>() {}

fn hr<T>() where for<'r#struct> T: Into<&'r#struct ()> {}

trait Foo<'r#struct> {}

trait Bar<'r#struct> {
    fn method(&'r#struct self) {}
    fn method2(self: &'r#struct Self) {}
}

fn labeled() {
    'r#struct: loop {
        break 'r#struct;
    }
}

fn main() {}

```

I saw this happen:

```
warning: the type of the `self` parameter does not need to be arbitrary
  --> src/main.rs:12:16
   |
12 |     fn method2(self: &'r#struct Self) {}
   |                ^^^^^^^^^^^^^^^^^^^^^ help: consider to change this parameter to: `&'struct self`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_arbitrary_self_type
   = note: `#[warn(clippy::needless_arbitrary_self_type)]` on by default
```

this will cause an error:
```
error: lifetimes cannot use keyword names
  --> src/main.rs:12:17
   |
12 |     fn method2(&'struct self) {}
   |                 ^^^^^^^

error: aborting due to 1 previous error
```

### Version

```text
rustc 1.85.0-nightly (8742e0556 2024-12-28)
binary: rustc
commit-hash: 8742e0556dee3c64f7144de2fb2e88936418865a
commit-date: 2024-12-28
host: x86_64-unknown-linux-gnu
release: 1.85.0-nightly
LLVM version: 19.1.6
```
